# Table of Contents

1. [Importing Libraries & Dataset](#importing-libraries--dataset)  
2. [Building EDA and Preparing for Data Visualization](#building-eda-and-preparing-for-data-visualization)  
3. [Visualizations](#visualizations)  
4. [Conclusion & Actionable Insights](#conclusion--actionable-insights)


# 1. Importing Libraries & Dataset and Plan

First of all, I need to import every library that I need for my notebook:

1. `pandas` for data manipulation, data cleaning, and exploratory data analysis (EDA).
2. `matplotlib` for data visualization (creating bar charts, histograms) to gain better pattern insights.
3. Download the dataset.

## Plan & Aims

1. **Exploratory Data Analysis (EDA):**
   - Perform a brief EDA to understand the dataset better.
   - Examine data types, identify missing values (NAN), and detect any anomalies (if they exist).

2. **Data Visualization:**
   - Create visualizations to gain insights into the data.
   - **Sales Overview:** Identify which product categories contribute the most to sales.
   - **Total Quantity and Value by Business:** Highlight the top purchasing businesses.

3. **Conclusions:**
   - Summarize findings based on the graphs and charts from the visualization section.

In [13]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [6]:
# Load the dataset(Excel file)
filePath = "/home/mine/Documents/DATA/Kwanza Tukule Data Analyst Assessment/Case Study Data - Read Only.xlsx"
xls = pd.ExcelFile(filePath)

# Display datasheet name to understand the file structure
xls.sheet_names

['case_study_data_2025-01-16T06_4']

In [8]:
# Load the dataset from the sheet
df = pd.read_excel(xls, sheet_name=xls.sheet_names[0])

# Display the first few rows
df.head()

DATE ANONYMIZED CATEGORY ANONYMIZED PRODUCT  \
0 2024-08-18 21:32:00        Category-106       Product-21f4   
1 2024-08-18 21:32:00        Category-120       Product-4156   
2 2024-08-18 21:32:00        Category-121       Product-49bd   
3 2024-08-18 21:32:00         Category-76       Product-61dd   
4 2024-08-18 21:32:00        Category-119       Product-66e0   

  ANONYMIZED BUSINESS ANONYMIZED LOCATION  QUANTITY  UNIT PRICE  
0       Business-de42       Location-1ba8         1       850.0  
1       Business-de42       Location-1ba8         2      1910.0  
2       Business-de42       Location-1ba8         1      3670.0  
3       Business-de42       Location-1ba8         1      2605.0  
4       Business-de42       Location-1ba8         5      1480.0

## Data Quality Assesment
I will now inpect the dataset.

1. **Missing Values** - Checking for null values.
2. **Duplicates** - Identifying repeated rows.
3. **Inconsistent Data Types** - Ensuring proper data types for analysis.

Let proceed.

In [19]:
# Check missing values
missing_values = df.isnull().sum()

# Check duplicates
duplicate_rows = df.duplicated().sum()

# Check the data types on eacch column
data_types = df.dtypes

# Compile the findings
data_report = pd.DataFrame({
    "Missing Values" : missing_values,
    "Duplicates" : [duplicate_rows] + [None] * (len (df.columns) - 1),
    "Data Types" : data_types
})



# Display the results
from IPython.display import display

display(data_report)





Missing Values  Duplicates      Data Types
DATE                              0      3524.0  datetime64[ns]
ANONYMIZED CATEGORY               0         NaN          object
ANONYMIZED PRODUCT                0         NaN          object
ANONYMIZED BUSINESS               0         NaN          object
ANONYMIZED LOCATION               0         NaN          object
QUANTITY                          0         NaN           int64
UNIT PRICE                        8         NaN         float64

## Data quality Findings.

1. **Missing Values** - No missing values were found in any column.
2. **Duplicates** - The data contains the 3524 duplicate rows. They need to be looked at. It is a good idea to review the **Date** duplicates to understand them.
3. **Inconsistent Data Types**
- The **Date** column is correctly formatted as `datetime64[ns]`
- All categorical fields are stored as `object`, **QUANTITY** as `int64` and **UNIT  PRICE** as `float64` which is expected.


In [21]:
# Display duplicate rows
duplicate_rows_df = df[df.duplicated(keep=False)]  # keep=False to show all duplicates

# Show few duplicate rows(10 rows)
duplicate_rows_df.head(10)

DATE ANONYMIZED CATEGORY ANONYMIZED PRODUCT  \
310 2024-05-17 20:27:00        Category-120       Product-1e80   
311 2024-05-17 20:27:00         Category-75       Product-8f75   
426 2024-01-31 19:34:00         Category-76       Product-e805   
526 2024-02-05 18:24:00        Category-120       Product-29ee   
570 2024-04-19 15:19:00         Category-75       Product-086d   
571 2024-04-19 15:19:00        Category-106       Product-21f4   
572 2024-04-19 15:19:00        Category-120       Product-4156   
641 2024-01-02 21:12:00        Category-119       Product-66e0   
777 2024-10-16 11:54:00        Category-108       Product-d8c2   
778 2024-10-16 11:54:00         Category-75       Product-e82e   

    ANONYMIZED BUSINESS ANONYMIZED LOCATION  QUANTITY  UNIT PRICE  
310       Business-9909       Location-689f         1      1970.0  
311       Business-9909       Location-689f         1      4020.0  
426       Business-bf85       Location-1ba8         5      7018.0  
526       Business-bfcd       Location-3e32         5      2290.0  
570       Business-b48e       Location-03fc         3      2090.0  
571       Business-b48e       Location-03fc         2       850.0  
572       Business-b48e       Location-03fc         3      2060.0  
641       Business-f9ff       Location-1979         1      1740.0  
777       Business-c569       Location-f37d         1      2960.0  
778       Business-c569       Location-f37d        10      2240.0